apiserver/src/main/java/org/dependencytrack/resources/v1/UserResource.java

```diff
@@ -256,6 +258,75 @@ public Response forceChangePassword(@FormParam("username") String username, @For
+            description = "<p>Requires permission <strong>ACCESS_MANAGEMENT</strong> or <strong>ACCESS_MANAGEMENT_READ</strong></p>"
+            + "<p><strong>Note:</strong> To retrieve additional subclass-specific information (such as ManagedUser fields like <code>suspended</code>, <code>forcePasswordChange</code>, etc.), "
+            + "you must specify the <code>type</code> query parameter (e.g., <code>type=managed</code>, <code>type=ldap</code>, or <code>type=oidc</code>). "
+            + "If <code>type</code> is omitted, only base User fields will be included in the response.</p>"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "A list of users or a single user (may be filtered by type and/or username)",
+                    headers = @Header(name = TOTAL_COUNT_HEADER, description = "The total number of managed users", schema = @Schema(format = "integer")),
+                    content = @Content(
+                        schema = @Schema(
+                            oneOf = {User.class, ManagedUser.class, LdapUser.class, OidcUser.class}
+                        ),
+                        array = @ArraySchema(
+                            schema = @Schema(oneOf = {User.class, ManagedUser.class, LdapUser.class, OidcUser.class})
+                        )
+                    )
+            ),
+            @ApiResponse(responseCode = "401", description = "Unauthorized")
+    })
+    @PermissionRequired({Permissions.Constants.ACCESS_MANAGEMENT, Permissions.Constants.ACCESS_MANAGEMENT_READ})
+    public Response getUsers(
+        @QueryParam("type") UserType type,
+        @QueryParam("username") String username) {
+        try (QueryManager qm = new QueryManager(getAlpineRequest())) {
+            final List<User> users;
+            if (type == null) {
+                users = qm.getAllUsers();
+                return Response.ok(users).header(TOTAL_COUNT_HEADER, users.size()).build();
+            }
+
+            Query<? extends User> query = qm.getPersistenceManager().newQuery(type.getUserClass());
+
+            if (username != null) {
+                query.setFilter("username == :username");
+                query.setNamedParameters(Map.of("username", username));
+            }
+
+            try {
+                users = List.copyOf(query.executeList());
+            } finally {
+                query.closeAll();
+            }
+
+            // failure
+            if(users == null) {
+                StringBuilder errStr = new StringBuilder();
+
+                errStr.append(username != null ? "User '%s'".formatted(username) : "Users");
+                errStr.append(type != null ? " of type '%s'".formatted(type) : " of all types");
+                errStr.append(" could not be found.");
+
+                return Response.status(Response.Status.NOT_FOUND).entity(errStr.toString()).build();
+            }
```

Comment:
To me, a single error string with the request parameters would be a cleaner solution

```suggestion
            if(users == null)
                return Response.status(Response.Status.NOT_FOUND)
                        .entity("No user(s) found for the given criteria [type=%s, username=%s]"
                                .formatted(type, username))
                        .build();
```